bee-identity/src/identity.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Contains the [`Identity`] type.
+
+use crate::keys::{Ed25519Keypair, Ed25519PublicKey};
+
+use std::fmt;
+
+/// Represents a local identity.
+#[derive(Clone, Debug)]
+pub struct Identity {
+    /// The identity alias.
+    alias: String,
+    /// An Ed25519 keypair.
+    keypair: Ed25519Keypair,
+    /// Whether the identity was newly generated.
+    is_new: bool,
```

Comment:
I don't know how to do this differently at the moment, but I feel like this is an implementation detail of `bee-node` that is leaking into this crate.

Reply:
Hmm, I see. I try to come up with a solution.

Reply:
I moved the migration stuff into `bee-identity` which I found was the cleanest solution. Whether an `identity` is new/fresh is now completely contained.